framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -670,6 +703,58 @@ protected void initializeReflectionResolution() {
      * @param root the new compilation unit to use
      */
     public void setRoot(@Nullable CompilationUnitTree root) {
+        setRoot(root, /*shouldCheckVisitor=*/ true);
+    }
+
+    /**
+     * Set the CompilationUnitTree that should be used.
+     *
+     * @param root the new compilation unit to use
+     * @param shouldCheckVisitor if true, run the visitor that verifies that the javac tree can be
+     *     visited with its corresponding JavaParser AST. The check only occurs if
```

Comment:
At first I thought this comment was explaining when `shouldCheckVisitor` is true.  But it is actually talking about a different boolean value.  What is the relationship between them, and why are both needed?  That is, why can't `shouldCheckeVisitor` be set according to `-AcheckJavaParserVisitor`?